packages/eslint-plugin/docs/rules/no-unsafe-type-assertion.mdx

```diff
@@ -0,0 +1,61 @@
+---
+description: 'Disallow type assertions that widen a type.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unsafe-type-assertion** for documentation.
+
+This rule forbids using type assertions to narrow a type, as it can lead to unsafe assumptions that bypass TypeScript’s type-checking.
```

Comment:
> unsafe assumptions

_What_ unsafe assumptions?

Someone reading this who isn't already familiar with the concepts wouldn't likely understand what it means. I think it'd even be easy to misinterpret this as saying _all_ type assertions are unsafe.

Could you add a section to this page explaining this for someone who doesn't already understand?

Reply:
I definitely agree! I've changed/expanded the opening paragraph to explain this better. Wdyt?